File: pinyougou_manager/pinyougou_manager_interface/src/main/java/com/pinyougou/manager/service/SpecificationService.java
package com.pinyougou.manager.service;

import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.SpecificationVo;
import com.pinyougou.pojo.TbSpecification;

import java.util.List;
import java.util.Map;

public interface SpecificationService {

    //查询列表
    PageResult search(Integer pageNum, Integer pageSize, TbSpecification tbSpecification);

    //添加规格及规格选项
    void add(SpecificationVo specificationVo);

    //根据规格ID查询规格数据及规格选项数据
    SpecificationVo findOne(Long id);

    //修改规格及规格选项
    void update(SpecificationVo specificationVo);

    //删除规格
    void dele(Long[] ids);

    //查询规格集合
    List<Map> findSpecList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/specificationService.js
app.service('specificationService',function ($http) {
    
    this.search=function (pageNum, pageSize, searchEntity) {
        return $http.post("../specification/search?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    };

    this.add=function (entity) {
        return $http.post('../specification/add',entity);
    };

    this.findOne=function(id){
        return $http.get("../specification/findOne?id="+id);
    };

    this.update=function(entity){
        return $http.post("../specification/update",entity);
    };

    this.dele=function (ids) {
        return $http.get('../specification/dele?ids='+ids);
    }

    this.findSpecList=function () {
        return $http.get('../specification/findSpecList');
    }

});